blocks/library/gallery/index.js

```diff
@@ -61,16 +70,11 @@ registerBlockType( 'core/gallery', {
 			query( 'div.blocks-gallery figure.blocks-gallery-image img', {
 				url: attr( 'src' ),
 				alt: attr( 'alt' ),
-			} ),
+			} ) || [],
```

Comment:
> I hadn’t realized however that the `attributes` property is evaluated only when parsing an existing block and not when creating a new one.

This is an interesting observation, and not one I'd internalized myself. I agree this would be quite confusing.

Attributes initialization in general is an area I think needs a considerable rethink. There's a handful of discussion issues and alternatives explorations that have been tried:

- #865
- #391
- #609
- #892
- #714 
- #914